src/ServiceControl.Monitoring/Http/Diagrams/TimingsAggregator.cs

```diff
@@ -21,22 +25,35 @@ public IEnumerable<MonitoredEndpoint> AggregateIntoLogicalEndpoints()
 
             var processingTimes = processingTimeStore.GetTimings(now);
             var criticalTimes = criticalTimeStore.GetTimings(now);
+            var queueLengths = queueLengthDataStore.GetQueueLengths(now);
 
-            var endpointNames = processingTimes.Concat(criticalTimes).Select(t => t.Id.EndpointName).Distinct();
+            var endpointNames = processingTimes.Concat(criticalTimes).Select(t => t.Id.EndpointName).Concat(queueLengths.Keys).Distinct();
 
             foreach (var endpointName in endpointNames)
             {
                 var instanceProcessingTime = processingTimes.Where(pt => pt.Id.EndpointName == endpointName).ToList();
                 var instanceCriticalTime = criticalTimes.Where(ct => ct.Id.EndpointName == endpointName).ToList();
+                var queueLength = queueLengths.FirstOrDefault(ql => ql.Key == endpointName);
 
+                var minDate = queueLength.Value?.Min(kvp => kvp.Key);
                 var monitoredEndpoint = new MonitoredEndpoint
                 {
                     Name = endpointName,
                     EndpointInstanceIds = new[] {instanceProcessingTime, instanceCriticalTime }
                         .SelectMany(t => t).Select(t => t.Id.InstanceId)
                         .Distinct().ToArray(),
                     ProcessingTime = AggregateTimings(instanceProcessingTime),
-                    CriticalTime = AggregateTimings(instanceCriticalTime)
+                    CriticalTime = AggregateTimings(instanceCriticalTime),
+                    QueueLength = new LinearMonitoredValues()
+                    {
+                        Average = queueLength.Value?.Values.Average(),
+                        Points = queueLength.Value?.OrderBy(kvp => kvp.Key)
```

Comment:
`queueLength.Value?.OrderBy(kvp => kvp.Key)` can be extracted to a variable and `ToArray`ed as it's used in `Points` as in `PointsAxisValues`.